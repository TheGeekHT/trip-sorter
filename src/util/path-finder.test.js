import PathFinder from './path-finder'
import trips from 'deals'

describe('path finder class', () => {
  test('get fastest direct trip', () => {
    const result = new PathFinder(trips, 'Istanbul', 'Athens', 'fastest').find();
    expect(result).toMatchSnapshot();
  });

  test('get cheapest direct trip', () => {
    const result = new PathFinder(trips, 'Istanbul', 'Athens', 'cheapest').find();
    expect(result).toMatchSnapshot();
  });

  test('get fastest trip', () => {
    const result = new PathFinder(trips, 'Kiev', 'Paris', 'fastest').find();
    expect(result).toMatchSnapshot();
  });

  test('get cheapest trip', () => {
    const result = new PathFinder(trips, 'Kiev', 'Paris', 'cheapest').find();
    expect(result).toMatchSnapshot();
  });

  test('check wrong departure/arrival city', () => {
    const result = new PathFinder(trips, 'K', 'Paris', 'cheapest').find();
    expect(result).toMatchSnapshot();
  });

  test('check if there is no trips data', () => {
    expect(() => {
      new PathFinder(undefined, 'Kiev', 'Paris', 'cheapest').find()
    }).toThrowErrorMatchingSnapshot();
  });

  test('unsupported trip type', () => {
    expect(() => {
      new PathFinder(trips, 'Istanbul', 'Athens', 'dadsasd').find();
    }).toThrowErrorMatchingSnapshot();
  });
});