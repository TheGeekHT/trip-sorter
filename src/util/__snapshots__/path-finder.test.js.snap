exports[`path finder class check if there is no trips data 1`] = `"No trips data received"`;

exports[`path finder class check wrong departure/arrival city 1`] = `Array []`;

exports[`path finder class get cheapest direct trip 1`] = `
Array [
  "BIA0515",
]
`;

exports[`path finder class get cheapest trip 1`] = `
Array [
  "BKB0600",
  "BBG0500",
  "BGM0645",
  "BMP0530",
]
`;

exports[`path finder class get fastest direct trip 1`] = `
Array [
  "TIA0330",
]
`;

exports[`path finder class get fastest trip 1`] = `
Array [
  "CKB0415",
  "TBG0300",
  "TGB0300",
  "TBP0400",
]
`;

exports[`path finder class unsupported trip type 1`] = `"Unsupported trip type"`;
